packages/connectors/connector-aliyun-dm/src/utils.ts

```diff
@@ -41,11 +49,13 @@ export const request = async (
   parameters: PublicParameters & Record<string, string>,
   accessKeySecret: string
 ) => {
-  const finalParameters: Record<string, string> = {
-    ...parameters,
-    SignatureNonce: String(Math.random()),
-    Timestamp: new Date().toISOString(),
-  };
+  const finalParameters = Object.fromEntries<string>(
+    Object.entries<string | undefined>({
+      ...parameters,
+      SignatureNonce: generateStandardId(),
+      Timestamp: formatDateString(new Date()),
+    }).filter((entry): entry is [string, string] => entry[1] !== undefined)
```

Comment:
What is this filter's use-case?

Reply:
Should drop undefined key-value pairs before generating signatures.

Reply:
why this function will receive undefined values?

Reply:
"From Alias" in config is optional.
<img width="613" alt="image" src="https://github.com/logto-io/logto/assets/15182327/297cecb1-573b-427d-9fa3-00f9376ef572">